### DISCUSSION TITLE
v3.1.1: Using the comma_separated_list with a pound sign

### DISCUSSION THREAD

**Discussion by leewilmott** [pyparsing/pyparsing#513] [General]
I hope someone can help me.  I'm using PyParsing v3.1.1.

I'm trying to separate a Comma Separated Values list and I'm having some difficulties.

This is what I'm trying to do:

I obviously start with this:
`import pyparsing`

...and then...
`>>> test = "Item1, Item2, Item3, £10.99, Item 5, \"Item 6a, Item 6b\""`
`>>> pyparsing.common.comma_separated_list.parseString(test).asList()`

The above returns the following:
`['Item1', 'Item2', 'Item3', '']`

...but I was expecting the result to be:
`['Item1', 'Item2', 'Item3', '£10.99', 'Item 5', 'Item 6a, Item 6b']`

If I remove the pound sign I receive different results.  

`>>> test = "Item1, Item2, Item3, 10.99, Item 5, \"Item 6a, Item 6b\""`
`>>> pyparsing.common.comma_separated_list.parseString(test).asList()`

which returns this:
`['Item1', 'Item2', 'Item3', '10.99', 'Item 5', '"Item 6a, Item 6b"']`

...but I was expecting the result to be:
`['Item1', 'Item2', 'Item3', '10.99', 'Item 5', 'Item 6a, Item 6b']`

So, I have two problems:
1) How can I include elements with pound signs?
2) How can I return the last element without the quotes?

Can anyone please help me?

**Comment by leewilmott** [comment]
I managed to find the answer to my own problem.

I looked at the code and worked out how DelimitedList works (by looking at [this](https://github.com/pyparsing/pyparsing/blob/master/pyparsing/common.py#L339)).

This allowed me to write the following code:

```
from pyparsing import Word, Opt, Combine, DelimitedList, Literal, unicode, White, FollowedBy, LineEnd, quoted_string, OneOrMore
from re import sub

def remove_double_quotes_around_string(string: str):
    remove_start_quotes = sub("^\"", "", string) # Remove start quotes
    remove_end_quotes = sub("\"$", "", remove_start_quotes) # Remove end quotes
    return remove_end_quotes

def get_csv_list_from_string(csv_string: str):
    word_obj = Word(unicode.Latin1.printables, exclude_chars=",")
    opt_obj = Opt(White(" \t") + ~FollowedBy(LineEnd() | ","))
    combine_obj = Combine(OneOrMore(~Literal(",") + ~LineEnd() + word_obj + opt_obj))
    delimited_list_obj = DelimitedList(Opt(quoted_string.copy() | combine_obj, default=""))
    csv_list = delimited_list_obj.parse_string(csv_string).as_list()
    return [remove_double_quotes_around_string(item) for item in csv_list]


my_string = 'Item1, Item2, Item3, £10.99, Item 5, "Item 6a, Item 6b"'

get_csv_list_from_string(my_string)
```

The above returns this:
`['Item1', 'Item2', 'Item3', '£10.99', 'Item 5', 'Item 6a, Item 6b']`
...which is exactly what I want!

Hopefully this might help someone who has the same requirement as me.

**Comment by ptmcg** [comment]
I'm glad you were able to work this out!
